services/api/service.go

```diff
@@ -1518,21 +1522,21 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}()
 
-	// Get the latest builder bid value from Redis.
-	latestBuilderValue, err := api.redis.GetBuilderLatestValue(payload.Slot(), payload.ParentHash(), payload.ProposerPubkey(), payload.BuilderPubkey().String())
+	// Grab floor bid value
+	floorBidValue, err := api.redis.GetFloorBidValue(payload.Slot(), payload.ParentHash(), payload.ProposerPubkey())
 	if err != nil {
-		log.WithError(err).Error("failed to get latest builder bid value from redis")
+		log.WithError(err).Error("failed to get floor bid value from redis")
 	} else {
-		bidHasHigherValue := payload.Value().Cmp(latestBuilderValue) == 1
+		isBidAboveFloor := payload.Value().Cmp(floorBidValue) == 1
 		log = log.WithFields(logrus.Fields{
-			"latestValue":          latestBuilderValue.String(),
-			"newBidHasHigherValue": bidHasHigherValue,
+			"floorBidValue":   floorBidValue.String(),
+			"isBidAboveFloor": isBidAboveFloor,
 		})
 
-		// Without cancellations, discard lower or similar value submissions.
-		if !isCancellationEnabled && !bidHasHigherValue {
-			log.Info("rejecting submission because without cancellation and not higher value than previous bid by this builder")
-			api.Respond(w, http.StatusAccepted, struct{ message string }{message: "ignoring submission because lower or equal value than previous bid"})
+		// Without cancellations, discard bids below floor value
+		if !isCancellationEnabled && !isBidAboveFloor {
```

Comment:
Can we relax this by removing `!isCancellationEnabled`?

The floor value is set from non-cancellable bids
but cancellable bids themselves still subject to the same floor.

In what scenario, cancellable bid below this floor will go to the proposer?

Reply:
That'd be great!

But what if 
- floor was 10
- last cancellable bid was 11
- cancels with bid value 5

With your suggestion applied, the 11 value bid wouldn't get cancelled? I guess we could put it into low prio queue though 🤔

Cc/ @michaelneuder

Reply:
You are right! We can't just remove this clause in if.

But on the other hand we can cancel this 11 bid without validating bid 5.

Reply:
that's smart!